masterlist_scraper.py

```diff
@@ -88,29 +107,54 @@
             continue
         for item in year_list:
             issue_url=item.get_attribute('href')
-            #Jstor_issue_text=item.text
-            #data=data.append(pd.Series([int(temp), issue_url], index=data.columns), ignore_index=True )
+            print(issue_url)
             issue_url_list.append(issue_url)
 
+            # Creating a tracker file
+            #url_item=pd.DataFrame({'url':[issue_url],'scraped':[0],'Journal':[journal]})
+            #issue_url_df=issue_url_df.concat(issue_url_df,url_item)
+    
+    #issue_url_df.to_csv("issue_url_df.csv")
+
     # loops through a dataframe of issue urls and captures metadata per issue
-    for issue_url in issue_url_list:
+    for issue_url in issue_url_list[issue_start:len(issue_url_list)]:
         time.sleep(5*random.random())     
         driver.get(issue_url)
 
-        # Download Citations
-        time.sleep(10)
+        try:
+            # Download Citations
 
-        WebDriverWait(driver,30).until(expected_conditions.element_to_be_clickable((By.XPATH,r".//toc-view-pharos-checkbox[@id='select_all_citations']/span[@slot='label']"))).click()
-        WebDriverWait(driver,30).until(expected_conditions.element_to_be_clickable((By.ID,"export-bulk-drop"))).click()
-        WebDriverWait(driver,30).until(expected_conditions.element_to_be_clickable((By.XPATH,r".//toc-view-pharos-dropdown-menu-item[@class='bibtex_bulk_export export_citations']"))).click()
+            WebDriverWait(driver,30).until(expected_conditions.element_to_be_clickable((By.XPATH,r".//toc-view-pharos-checkbox[@id='select_all_citations']/span[@slot='label']"))).click()
+            WebDriverWait(driver,30).until(expected_conditions.element_to_be_clickable((By.ID,"export-bulk-drop"))).click()
 
-        time.sleep(10)
+            time.sleep(10)
+
+            #download_metadata=driver.find_element(By.XPATH,r".//toc-view-pharos-dropdown-menu-item[@class='bibtex_bulk_export export_citations']")
+            download_metadata=driver.find_element(By.XPATH,r"//*[@id='bulk-citation-dropdown']/toc-view-pharos-dropdown-menu-item[5]")
+            print(download_metadata.size)
+
+            driver.implicitly_wait(5)
+            
+            ActionChains(driver).move_to_element(download_metadata).perform()
+        
+            driver.implicitly_wait(10)
+            #WebDriverWait(driver,30).until(expected_conditions.element_to_be_clickable((By.XPATH,r"//*[@id='bulk-citation-dropdown']/toc-view-pharos-dropdown-menu-item[5]"))).click()
+            driver.execute_script("arguments[0].click();", download_metadata)
+            
+        except Exception as e:
+            print(e)
+            input()
 
         old_name=os.path.join(directory,'citations.txt')
         new_name=os.path.join(directory,issue_url.split("/")[-1]+'.txt')
+
+        # Wait for download to complete
+        while not os.path.isfile(old_name):
```

Comment:
This loop is potentially infinite if the file is never successfully downloaded. Consider adding a maximum_duration constant that signifies the longest time we are willing to wait for a download before failing.